nemo/collections/nlp/data/machine_translation/preproc_mt_data.py

```diff
@@ -366,6 +348,91 @@ def preprocess_parallel_dataset(
 
         return tar_file_paths, metadata_path
 
+    @staticmethod
+    def _get_num_lines(filename):
+        with open(filename) as f:
+            for i, l in enumerate(f):
+                pass
+        return i + 1
+
+    @staticmethod
+    def _get_lines_partition(num_lines, lines_per_dataset_fragment):
+        # create partition based on fragment size
+        fragment_indices = []
+        for i in range(0, num_lines, lines_per_dataset_fragment):
+            fragment_indices.append([i, i + lines_per_dataset_fragment])
+        # remove last indices if needed
+        if fragment_indices[-1][1] >= num_lines:
+            fragment_indices.pop()
+        return fragment_indices
+
+    @staticmethod
+    def _process_fragment(
+        src_filename,
+        tgt_filename,
+        lines_indices,
+        out_dir,
+        num_batches_per_tarfile,
+        clean,
+        max_seq_length,
+        min_seq_length,
+        tokens_in_batch,
+        encoder_tokenizer_name,
+        encoder_tokenizer_model,
+        encoder_bpe_dropout,
+        encoder_model_name,
+        decoder_tokenizer_name,
+        decoder_tokenizer_model,
+        decoder_bpe_dropout,
+        decoder_model_name,
+        pkl_file_prefix,
+    ):
+        start = lines_indices[0]
+        stop = lines_indices[1]
+
+        # write lines in partition to temporary files to be consumed by write_parallel_batches_to_tarfiles
+        tmp_f_src = tempfile.NamedTemporaryFile(delete=False, mode='w')
+        tmp_f_tgt = tempfile.NamedTemporaryFile(delete=False, mode='w')
+
+        with open(src_filename, 'r') as src_in, open(tgt_filename) as tgt_in:
+            src_line = next(src_in)
```

Comment:
Why do you call next on the file before enumerate? Won't that discard the first line before you start enumerating the file?

Reply:
I believe that gives you the first line. I can double check.

Reply:
Done.